CLI app: sync accuracy of the report \w Linguist (#214)
src-d/enry: README.md

Divergences from linguist
------------

`enry` [CLI tool](#cli) does *not* require a full Git repository to be present in the filesystem in order to report languages.

Using [linguist/samples](https://github.com/github/linguist/tree/master/samples)
as a set for the tests, the following issues were found:

* [Heuristics for ".es" extension](https://github.com/github/linguist/blob/e761f9b013e5b61161481fcb898b59721ee40e3d/lib/linguist/heuristics.yml#L103) in JavaScript could not be parsed, due to unsupported backreference in RE2 regexp engine

> * As of (Linguist v5.3.2)[https://github.com/github/linguist/releases/tag/v5.3.2] it is using [flex-based scanner in C for tokenization](https://github.com/github/linguist/pull/3846). Enry stil uses [extract_token](https://github.com/github/linguist/pull/3846/files#diff-d5179df0b71620e3fac4535cd1368d15L60) regex-based algorithm. See [#193](https://github.com/src-d/enry/issues/193).

Reviewer: ```suggestion
* As of (Linguist v5.3.2)[https://github.com/github/linguist/releases/tag/v5.3.2] it is using [flex-based scanner in C for tokenization](https://github.com/github/linguist/pull/3846). Enry still uses [extract_token](https://github.com/github/linguist/pull/3846/files#diff-d5179df0b71620e3fac4535cd1368d15L60) regex-based algorithm. See [#193](https://github.com/src-d/enry/issues/193).
```

* [Heuristics for ".es" extension](https://github.com/github/linguist/blob/e761f9b013e5b61161481fcb898b59721ee40e3d/lib/linguist/heuristics.yml#L103) in JavaScript could not be parsed, due to unsupported backreference in RE2 regexp engine

> * As of (Linguist v5.3.2)[https://github.com/github/linguist/releases/tag/v5.3.2] it is using [flex-based scanner in C for tokenization](https://github.com/github/linguist/pull/3846). Enry stil uses [extract_token](https://github.com/github/linguist/pull/3846/files#diff-d5179df0b71620e3fac4535cd1368d15L60) regex-based algorithm. [#193](https://github.com/src-d/enry/issues/193)

Reviewer: It may be worth mentioning explicitly that we plan to fix this, e.g.

> We plan to update Enry to match Linguist, tracked in [#193](https://github.com/src-d/enry/issues/193).

Author: That is true for all the things that have an issue number, 👍 for mentioning that explicitly though.

> * Bayesian classifier cann't distinguish "SQL" vs "PLpgSQL. [#194](https://github.com/src-d/enry/issues/194)

Reviewer: ```suggestion
* Bayesian classifier can't distinguish "SQL" from "PLpgSQL. See [#194](https://github.com/src-d/enry/issues/194).
```

> * Dection of [generated files](https://github.com/github/linguist/blob/bf95666fc15e49d556f2def4d0a85338423c25f3/lib/linguist/generated.rb#L53) is not supported yet.

Reviewer: ```suggestion
* Detection of [generated files](https://github.com/github/linguist/blob/bf95666fc15e49d556f2def4d0a85338423c25f3/lib/linguist/generated.rb#L53) is not supported yet.
```

>  (Thus they are not exclued from CLI output) [#213](https://github.com/src-d/enry/issues/213)

Reviewer: ```suggestion
 (Thus they are not excluded from CLI output). See [#213](https://github.com/src-d/enry/issues/213).
```

* XML detection strategy is not implemented. [#192](https://github.com/src-d/enry/issues/192)

> * Overriding languaes and types though `.gitattributes` is not yet supported. [#18](https://github.com/src-d/enry/issues/18)

Reviewer: ```suggestion
* Overriding languages and types though `.gitattributes` is not yet supported. See [#18](https://github.com/src-d/enry/issues/18).
```

> * enry CLI output does NOT exclude `.gitignore`ed files and submodel dirs as linguist does

Reviewer: I suggest either:

> `enry` CLI output …

or

> Enry CLI output …

depending on whether we mean the name of the project or the name of the tool.
